src/main/src/main/java/org/geoserver/security/impl/ServiceAccessRuleDAO.java

```diff
@@ -113,7 +113,8 @@ ServiceAccessRule parseServiceAccessRule(String ruleKey, String ruleValue) {
     @Override
     protected Properties toProperties() {
         Properties props = new Properties();
-        for (ServiceAccessRule rule : rules) {
+        TreeSet<ServiceAccessRule> sortedRules = new TreeSet<>(rules);
```

Comment:
You could use a sorted stream instead

Reply:
Regardless, does it actually work? Properties extends HashTable, putting data in it may scramble the order anyways?